modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -205,18 +198,22 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
 #endif
         if (backendId == DNN_BACKEND_INFERENCE_ENGINE_NGRAPH)
         {
-            return !computeMaxIdx && type != STOCHASTIC;
+            return !computeMaxIdx && type != STOCHASTIC && kernel_size.size() > 1;
         }
-        else if (backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE)
+        else if (backendId == DNN_BACKEND_OPENCV)
         {
             if (kernel_size.size() == 3)
-                return (backendId == DNN_BACKEND_OPENCV && preferableTarget == DNN_TARGET_CPU);
-            if (kernel_size.empty() || kernel_size.size() == 2)
-                return backendId == DNN_BACKEND_OPENCV ||
-                       (backendId == DNN_BACKEND_HALIDE && haveHalide() &&
-                           (type == MAX || (type == AVE && !pad_t && !pad_l && !pad_b && !pad_r)));
-            else
-                return false;
+                return preferableTarget == DNN_TARGET_CPU;
+            if (kernel_size.size() == 1)
+                return (preferableTarget == DNN_TARGET_CPU ||
+                        preferableTarget == DNN_TARGET_OPENCL ||
+                        preferableTarget == DNN_TARGET_OPENCL_FP16);
```

Comment:
Do we have other targets?

Reply:
> Do we have other targets?

I just try to revert my changes to see why the tests broke. Everything was fine until I changed logic for support checks. See, for instance, https://pullrequest.opencv.org/buildbot/builders/precommit_custom_mac/builds/1438